builtin/providers/google/resource_google_project.go

```diff
@@ -55,87 +65,195 @@ func resourceGoogleProject() *schema.Resource {
 	}
 }
 
-// This resource supports creation, but not in the traditional sense.
-// A new Google Cloud Project can not be created. Instead, an existing Project
-// is initialized and made available as a Terraform resource.
 func resourceGoogleProjectCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	project, err := getProject(d, config)
+	project := &cloudresourcemanager.Project{
+		Name:      d.Get("name").(string),
+		ProjectId: d.Get("project_id").(string),
+		Parent: &cloudresourcemanager.ResourceId{
+			Id:   d.Get("org_id").(string),
+			Type: "organization",
+		},
+	}
+	op, err := config.clientResourceManager.Projects.Create(project).Do()
 	if err != nil {
-		return err
+		return fmt.Errorf("Error creating project %s (%s): %s.", project.ProjectId, project.Name, err)
 	}
 
-	d.SetId(project)
-	if err := resourceGoogleProjectRead(d, meta); err != nil {
+	// Wait for the operation to complete
+	waitErr := resourceManagerOperationWait(config, op, "project to create")
+	if waitErr != nil {
+		// The resource didn't actually create
+		d.SetId("")
+		return waitErr
+	}
+
+	// Set the resourc ID
+	d.SetId(project.ProjectId)
+
+	// Enable services
+	if err = enableServices(d, config, project.ProjectId); err != nil {
 		return err
 	}
 
-	// Apply the IAM policy if it is set
-	if pString, ok := d.GetOk("policy_data"); ok {
-		// The policy string is just a marshaled cloudresourcemanager.Policy.
-		// Unmarshal it to a struct.
-		var policy cloudresourcemanager.Policy
-		if err = json.Unmarshal([]byte(pString.(string)), &policy); err != nil {
-			return err
-		}
+	// Set IAM policy
+	return setProjectIamPolicy(d, config, project.ProjectId)
+}
 
-		// Retrieve existing IAM policy from project. This will be merged
-		// with the policy defined here.
-		// TODO(evanbrown): Add an 'authoritative' flag that allows policy
-		// in manifest to overwrite existing policy.
-		p, err := getProjectIamPolicy(project, config)
-		if err != nil {
-			return err
+func resourceGoogleProjectRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	pid := d.Id()
+	// Read the project
+	p, err := config.clientResourceManager.Projects.Get(pid).Do()
+	if err != nil {
+		if v, ok := err.(*googleapi.Error); ok && v.Code == http.StatusNotFound {
+			return fmt.Errorf("Project %q does not exist.", pid)
 		}
-		log.Printf("[DEBUG] Got existing bindings from project: %#v", p.Bindings)
+		return fmt.Errorf("Error checking project %q: %s", pid, err)
+	}
 
-		// Merge the existing policy bindings with those defined in this manifest.
-		p.Bindings = mergeBindings(append(p.Bindings, policy.Bindings...))
+	d.Set("number", strconv.FormatInt(int64(p.ProjectNumber), 10))
```

Comment:
Ahh, so this would read the actual state from the APIs (which accounts for the possibility that non-computed values were changed out-of-band?) Makes sense, will correct.